src/Mod/Path/PathScripts/PathCircularHoleBaseGui.py

```diff
@@ -29,6 +29,10 @@
 
 from PySide import QtCore, QtGui
 
+import sys
+if sys.version_info.major >= 3:
+    xrange = range
```

Comment:
Do you even need `xrange`? In Python 2 it was necessary to create just the iterator and not a list, and thus speed up loops. But unless your loop has many thousands of elements, I don't think the speed up is going to be significant.

I'd just change all `xrange` for `range`.